tsl/src/compression/compression.c

```diff
@@ -2790,20 +3181,53 @@ decompress_batches_for_update_delete(Chunk *chunk, List *predicates)
 		scankeys =
 			build_update_delete_scankeys(&decompressor, filters, &num_scankeys, &null_columns);
 	}
-	if (decompress_batches(&decompressor,
+	if (index_filters)
+	{
+		List *ordered_index_filters = NIL;
+		Relation matching_index_rel = find_matching_index(comp_chunk_rel, index_filters);
+		Assert(matching_index_rel);
+		ordered_index_filters = fix_and_reorder_index_filters(comp_chunk_rel,
+															  matching_index_rel,
+															  segmentby_predicates,
+															  index_filters);
+		index_scankeys = build_index_scankeys(decompressor.in_rel,
+											  matching_index_rel,
+											  ordered_index_filters,
+											  &num_index_scankeys,
+											  estate);
+		if (index_scankeys)
```

Comment:
under which circumstance would index_scankeys be NULL

Reply:
Fixed.